### PR TITLE
Test mode implementation, #24

### PR DESCRIPTION
**Pull request by pirogoeth**
Super basic implementation of a "test mode" allowing users to easily stub calls that send emails.

Initially, it is just for stubbing the message sending routine, but can be expanded to stub other API calls if necessary.

Test mode is also tested. :joy:

*EDIT*: Version has also been bumped to 1.1.2 for this feature.

### CODE REVIEW

**Comment by pirogoeth** [comment]
@sixfeetover @michaelrkn Any ideas on this?

**Comment by michaelrkn** [comment]
This looks great! Thanks @pirogoeth!

File: lib/mailgun/response.rb [hunk 1]
```diff
@@ -10,6 +10,25 @@ class Response
...
+        }
+      end
+
+      if h.key? :code then
+        resp.define_singleton_method(:code) {
+          h[:code]
+        }
+      end
+
+      self.new resp
```
Comment (lukebooth) [inline lib/mailgun/response.rb]: Can I suggest using an OpenStruct here? Then _if_ there ever was a time where you wanted more than `body` and `code` methods, you'll get them for free.

```ruby
require 'ostruct`
self.new OpenStruct.new(h)
```
Whatever key values you pass into the `from_hash` method will be callable without checking if the key exists and defining a method for it if it does.

**Comment by sixfeetover** [comment]
This looks great, thank you. 👍

File: lib/mailgun/client.rb [hunk 2]
```diff
@@ -12,13 +12,36 @@ class Client
     def initialize(api_key = Mailgun.api_key,
                    api_host = 'api.mailgun.net',
                    api_version = 'v3',
-                   ssl = true)
+                   ssl = true,
+                   test_mode = false)
```
Comment (kpandya91) [inline lib/mailgun/client.rb]: Should this default to `Mailgun.test_mode`?

Comment (pirogoeth) [inline lib/mailgun/client.rb]: @kpandya91 So actually that's a really good point. It looks like `Mailgun.test_mode` was defined in an earlier commit but was never actually used? So this actually could make use of that field. I'll dig through the code again to make sure it's unused and add that as the default value. Thanks!

**Pull request merged**